src/scan.c

```diff
@@ -239,10 +245,11 @@ static int scan_module(struct context_data *ctx, int ep, int chain)
 	    pdelay = 0;
 
 	    for (chn = 0; chn < mod->chn; chn++) {
-		if (row >= mod->xxt[mod->xxp[pat]->index[chn]]->rows)
+		if (row >= tracks[chn]->rows)
 		    continue;
 
-		event = &EVENT(mod->xxo[ord], chn, row);
+		//event = &EVENT(mod->xxo[ord], chn, row);
+		event = &tracks[chn]->event[row];
```

Comment:
Can't think straight right now - we're no longer referencing mod->xxo here anymore?

Reply:
Right— `mod->xxo[ord]` is stored to `pat` at line 149 and neither `ord` nor `pat` are modified after that part of the loop. This means the track pointers can be determined once instead of every single time an event is read. This is what `EVENT()` does, for reference:
```C
#define TRACK_NUM(a,c)	m->mod.xxp[a]->index[c]
#define EVENT(a,c,r)	m->mod.xxt[TRACK_NUM((a),(c))]->event[r]
```
Every single event it was dereferencing the pattern to get an index, then dereferencing the `mod->xxt` to get the track pointer, which is plenty of opportunities for a paranoid compiler to stick in explicit loads because it doesn't know `m->scan_cnt` and `p->sequence_control` don't somehow alias these.

Reply:
I see, OK then.